infrastructure/io/src/main/java/tech/pegasys/teku/infrastructure/io/SyncDataAccessor.java

```diff
@@ -15,18 +15,54 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.AtomicMoveNotSupportedException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
 import java.nio.file.StandardOpenOption;
 import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.infrastructure.exceptions.InvalidConfigurationException;
 
 /**
  * A wrapper class for reading and writing files, ensuring that the contents is immediately flushed
  * to disk.
  */
 public class SyncDataAccessor {
 
+  private static final Logger LOG = LogManager.getLogger();
+  private boolean atomicFileMoveSupport;
+
+  SyncDataAccessor(final boolean atomicFileMoveSupport) {
+    this.atomicFileMoveSupport = atomicFileMoveSupport;
+  }
+
+  public static SyncDataAccessor create(final Path path) {
+    boolean atomicFileMoveSupport = false;
+    final Path absolutePath = path.toAbsolutePath();
+    final Path tmpFile = Paths.get(absolutePath.toString(), "_temp.tmp");
```

Comment:
```suggestion
    final Path tmpFile = path.resolve("syncWriteTest.tmp");
```

The filename change is because some filesystems can be picky about characters at the start of the filename and there's a chance the file will be left around so having a name that is meaningful makes it easier to understand why it's there.